pvlib/iam.py

```diff
@@ -940,3 +942,383 @@ def schlick_diffuse(surface_tilt):
+
+    elif target_name == "physical":
+        L, n = optimize_result.x
+        # have to unpack order because search order may be different
+        if L > n:
+            L, n = n, L
+        target_params = {'n': n, 'K': 4, 'L': L}
+
+    return target_params
+
+
+def _min_scipy():
+    '''convert and fit require scipy>=1.5.0
+    '''
+    from scipy import __version__
+    major, minor = __version__.split('.')[:2]
+    if (int(major) >= 1) & (int(minor) >= 5):
+        return True
+    else:
+        warnings.warn('iam.convert and iam.fit require scipy>=1.5.0')
+        return False
+
+
+def convert(source_name, source_params, target_name, weight=_sin_weight,
+            fix_n=True, xtol=None):
+    """
+    Convert a source IAM model to a target IAM model.
+
+    Parameters
+    ----------
+    source_name : str
+        Name of the source model. Must be ``'ashrae'``, ``'martin_ruiz'``, or
+        ``'physical'``.
+
+    source_params : dict
+        A dictionary of parameters for the source model.
+
+            If source model is ``'ashrae'``, the dictionary must contain
+            the key ``'b'``.
+
+            If source model is ``'martin_ruiz'``, the dictionary must
+            contain the key ``'a_r'``.
+
+            If source model is ``'physical'``, the dictionary must
+            contain the keys ``'n'``, ``'K'``, and ``'L'``.
+
+    target_name : str
+        Name of the target model. Must be ``'ashrae'``, ``'martin_ruiz'``, or
+        ``'physical'``.
+
+    weight : function, optional
+        A single-argument function of AOI (degrees) that calculates weights for
+        the residuals between models. Must return a float or an array-like
+        object. The default weight function is :math:`f(aoi) = 1 - sin(aoi)`.
+
+    fix_n : bool, default True
+        A flag to determine which method is used when converting from the
+        ASHRAE model to the physical model.
+
+        When ``source_name`` is ``'ashrae'`` and ``target_name`` is
+        ``'physical'``, if `fix_n` is ``True``,
+        :py:func:`iam.convert` will fix ``n`` so that the returned physical
+        model has the same x-intercept as the inputted ASHRAE model.
+        Fixing ``n`` like this improves the fit of the conversion, but often
+        returns unrealistic values for the parameters of the physical model.
+        If more physically meaningful parameters are wanted,
+        set `fix_n` to False.
+
+    xtol : float, optional
+        Passed to scipy.optimize.minimize.
+
+    Returns
+    -------
+    dict
+        Parameters for the target model.
+
+            If target model is ``'ashrae'``, the dictionary will contain
+            the key ``'b'``.
+
+            If target model is ``'martin_ruiz'``, the dictionary will
+            contain the key ``'a_r'``.
+
+            If target model is ``'physical'``, the dictionary will
+            contain the keys ``'n'``, ``'K'``, and ``'L'``.
+
+    Note
+    ----
```

Comment:
I know the reference explains everything, but being a little more loquacious here wouldn't hurt.  Same in the other function's note.

Reply:
Some hints what you hoped to see here?

Reply:
No, it just struck me as terse.  Don't worry about it.